src/test/java/org/passay/doc/ClassGenerator.java

```diff
@@ -0,0 +1,312 @@
+   * @param  path  to write files to
+   *
+   * @throws  IOException  if the source cannot be downloaded, unzipped and read
+   */
+  public ClassGenerator(final String url, final String path)
+    throws IOException
+  {
+    // download the zipped source
+    download(url, path + "/source.zip");
+
+    // unzip the source
+    unzip(path + "/source.zip", path + "/doc-sources");
+
+    // read the source files
+    final Path sourceDir = Paths.get(path + "/doc-sources");
+    Files.walkFileTree(sourceDir, new SimpleFileVisitor<Path>() {
+      @Override
+      public FileVisitResult visitFile(final Path file, final BasicFileAttributes attrs)
+        throws IOException
+      {
+        if (attrs.isRegularFile()) {
+          final String content = new Scanner(file.toFile()).useDelimiter("\\Z").next();
+          final String name = file.getName(file.getNameCount() - 2).toString();
+          final List<String> l;
+          if (sections.containsKey(name)) {
+            l = sections.get(name);
+          } else {
+            l = new ArrayList<>();
+          }
+          l.add(content);
+          sections.put(name, l);
+        }
+        return FileVisitResult.CONTINUE;
+      }
+    });
+  }
+
+
+  /**
+   * Download a file.
+   *
+   * @param  url  to download
+   * @param  destination  to write the file at
+   *
+   * @throws  IOException  if an error occurs
+   */
+  private static void download(final String url, final String destination)
+    throws IOException
+  {
+    final URL download = new URL(url);
+    final ReadableByteChannel rbc = Channels.newChannel(download.openStream());
+    final FileOutputStream fos = new FileOutputStream(destination);
+    fos.getChannel().transferFrom(rbc, 0, Long.MAX_VALUE);
+    fos.flush();
+    rbc.close();
+  }
+
+
+  /**
+   * Unzip a file.
+   *
+   * @param  file  to unzip
+   * @param  destination  to expand the file at
+   *
+   * @throws  IOException  if an error occurs
+   */
+  private static void unzip(final String file, final String destination)
+    throws IOException
+  {
+    final File directory = new File(destination);
+    // if the output directory doesn't exist, create it
+    if (!directory.exists()) {
+      directory.mkdirs();
+    }
+    // buffer for read and write data to file
+    final byte[] buffer = new byte[BUFFER_SZIE];
+    final ZipInputStream zipInput = new ZipInputStream(new FileInputStream(file));
+    ZipEntry entry = zipInput.getNextEntry();
+    try {
+      while (entry != null) {
+        final String entryName = entry.getName();
+        final File f = new File(destination + File.separator + entryName);
+        // create the directories of the zip directory
+        if (entry.isDirectory()) {
+          final File newDir = new File(f.getAbsolutePath());
+          if (!newDir.exists()) {
+            newDir.mkdirs();
+          }
+        } else {
+          try (FileOutputStream fOutput = new FileOutputStream(f)) {
```

Comment:
After review of the `dictionary` package, which I assumed had similar needs, and other use cases in this class, I think stream handling utilities would provide little additional value.